core/domain/platform_parameter_domain.py

```diff
@@ -650,24 +697,25 @@ def feature_stage(self):
         """
         return self._feature_stage
 
-    def validate(self):
+    # We are comparing the same types
+    def validate(self) -> None:
         """Validates the PlatformParameter domain object."""
         if re.match(self.PARAMETER_NAME_REGEXP, self._name) is None:
             raise utils.ValidationError(
                 'Invalid parameter name \'%s\', expected to match regexp '
                 '%s.' % (self._name, self.PARAMETER_NAME_REGEXP))
 
-        if self._data_type not in self.DATA_TYPE_PREDICATES_DICT:
+        if self._data_type not in self.DATA_TYPE_PREDICATES_DICT: # type: ignore[comparison-overlap]
             raise utils.ValidationError(
                 'Unsupported data type \'%s\'.' % self._data_type)
 
-        predicate = self.DATA_TYPE_PREDICATES_DICT[self.data_type]
-        if not predicate(self._default_value):
+        predicate = self.DATA_TYPE_PREDICATES_DICT[self.data_type] # type: ignore[index]
+        if not predicate(self._default_value): # type: ignore[no-untyped-call]
```

Comment:
Why is this ignore needed?

Reply:
the method predicate is not typed yet and is in anoter file (which we didn't find)